infra/Pos/Slotting/Util.hs

```diff
@@ -88,9 +72,21 @@ getSlotStartEmpatically
 getSlotStartEmpatically slot =
     getSlotStart slot >>= maybeThrow (SEUnknownSlotStart slot)
 
+-- | Get last epoch empatically, which means
+-- that function throws exception when last epoch is unknown.
+getLastEpochEmpatically
+    :: (MonadSlotsData m, MonadThrow m)
+    => m EpochSlottingData
+getLastEpochEmpatically = do
+    lastEpoch   <- getEpochLastIndex    
+    getEpochSlottingData lastEpoch >>= maybeThrow (SEUnknownLastEpoch lastEpoch)
+
 -- | Get last known slot duration.
 getLastKnownSlotDuration :: MonadSlotsData m => m Millisecond
-getLastKnownSlotDuration = esdSlotDuration . sdLast <$> getSlottingData
+getLastKnownSlotDuration = do
+  eli <- getEpochLastIndex
+  -- AJ: TODO: GOOD DEFAULT VALUE FOR SLOT DURATION?
+  maybe 0 esdSlotDuration <$> getEpochSlottingData eli
```

Comment:
This function should use `getLastEpoch` and then it will not need a default value.